### PR TITLE
removed not needed recursive call to the destroy on replicatedmap

### LINKED ISSUE #6754: replicatedmap.destroy() goes into an infinite loop
`destroyDistributedObject()` method on `ReplicatedMapService` iterates over partition containers and record stores and destroys them. While destroying, record store calls `destroyDistributedObject()` which leads to an infinite loop.

### PR DESCRIPTION
**Pull request by eminn**
Fixes #6754

### CODE REVIEW

**Comment by eminn** [comment]
verify

**Comment by eminn** [comment]
run-lab-run

File: hazelcast/src/test/java/com/hazelcast/replicatedmap/ReplicatedMapTest.java [hunk 1]
```diff
@@ -1030,6 +1031,18 @@ public void testSizeAfterRemove() throws Exception {
...
     }
 
+
+    @Test
+    public void testDestroy() throws Exception {
+        HazelcastInstance instance = createHazelcastInstance();
+        ReplicatedMap<Object, Object> replicatedMap = instance.getReplicatedMap(randomName());
+        replicatedMap.put(1, 1);
+        replicatedMap.destroy();
+        Collection<DistributedObject> objects = instance.getDistributedObjects();
```
Comment (pveentjer) [inline hazelcast/src/test/java/com/hazelcast/replicatedmap/ReplicatedMapTest.java]: Add an empty line between 'doing' and 'asserting'. This style is recommended on the wiki to make it easier to see what a test is doing.

**Comment by pveentjer** [comment]
I don't see any reference to a bug. If this is a cleanup, it needs to be parked till 3.7.

File: hazelcast/src/test/java/com/hazelcast/replicatedmap/ReplicatedMapTest.java (hunk 1 above)
Comment (eminn) [inline hazelcast/src/test/java/com/hazelcast/replicatedmap/ReplicatedMapTest.java]: Will make a direct commit after merge

**Comment by pveentjer** [comment]
After a discussion with Emin.

If there are P partitions, then the total number of destroy calls is going to be P+P*P. 

This can become a performance problem if P is large enough. So we can consider this a performance bug and it is fine to get it merged.

**Pull request merged**